python/pyarrow/dataset.py

```diff
@@ -210,6 +211,26 @@ def partitioning(schema=None, field_names=None, flavor=None,
             raise ValueError(
                 "For the default directory flavor, need to specify "
                 "a Schema or a list of field names")
+    if flavor == "filename":
```

Comment:
Can you update the docstring to mention this possibility?

Reply:
Made the change.